src/pybind/mgr/call_home_agent/module.py

```diff
@@ -0,0 +1,322 @@
+"""
+
+from typing import List, Any, Tuple, Dict, Optional, Set, Callable
+from datetime import datetime
+import json
+import requests
+import asyncio
+import os
+
+from mgr_module import Option, CLIReadCommand, MgrModule, HandleCommandResult, NotifyType
+
+from .config import get_settings
+from .options import CHES_ENDPOINT, INTERVAL_INVENTORY_REPORT_MINUTES, INTERVAL_PERFORMANCE_REPORT_MINUTES
+
+# Module with the functions to be used in telemetry and in CCHA for retrieving information from the ceph cluster
+from .dataCollectors import inventory
+
+reports_header = {
+    "agent": "RedHat_Marina_firmware_agent",
+    "api_key": "",
+    "private_key": "",
+    "target_space": "dev",
+    "asset": "ceph",
+    "asset_id": "",
+    "asset_type": "RedHatMarine",
+    "asset_vendor": "IBM",
+    "asset_virtual_id": "",
+    "country_code": "",
+    "event_id": "",
+    "event_time": "",
+    "event_time_ms": 0,
+    "local_event_time": "",
+    "software_level": {
+        "name": "ceph_software",
+        "vrmf": "8.3.0.1"
+    },
+    "type": "eccnext_apisv1s",
+    "version": "1.0.0.1",
+    "analytics_event_source_type": "asset_event",
+    "analytics_type": "ceph",
+    "analytics_instance": "",
+    "analytics_virtual_id": "",
+    "analytics_group": "Storage",
+    "analytics_category": "RedHatMarine",
+    "events": []
+}
+
+# TODO: report functions providing the json paylod should be imported from the data collectors module
+
+
+def report_inventory() -> str:
+    """
+    Produce the inventory report
+    """
+    dt = datetime.timestamp(datetime.now())
+    # Set timestamps
+    reports_header['event_time'] = datetime.fromtimestamp(dt).strftime("%Y-%m-%d %H:%M:%S")
+    reports_header['event_time_ms'] = str(int(dt))
+    reports_header['local_event_time'] = datetime.fromtimestamp(dt).strftime("%a %b %d %H:%M:%S %Z")
+
+    # Set event id
+    reports_header['event_id'] = "IBM_chc_event_RedHatMarine_ceph_{}_daily_report_{}".format(
+                                            reports_header['asset_virtual_id'],
+                                            reports_header['event_time_ms'])
+
+    # return json.dumps({**reports_header, **inventory()}) <---- TODO ADD INVENTORY (from datacollector) as event
+    return json.dumps(reports_header)
+
+
+def report_performance() -> str:
+    """
+    TODO: temporally uses inventory report
+    """
+    return report_inventory()
+
+
+class Report:
+    def __init__(self, name: str, fn: Callable[[], str], url: str, minutes_interval: int):
+        self.name = name                       # name of the report
+        self.fn = fn                           # function used to retrieve the data
+        self.url = url                         # url to send the report
+        self.interval = minutes_interval * 60  # interval to send the report (seconds)
+
+    def print(self) -> str:
+        if reports_header['private_key'] and reports_header['api_key']:
+            return self.fn()
```

Comment:
For a print, shouldn't we try to mask the sensitive fields? (At least the key fields)
We could also introduce a toggle to not mask the fields that is disabled by default...

Reply:
We are going to use a class derived from  named tuple as @epuertat suggested